From 695ae7ce37534156f5bc553b42de172dccd4f74c Mon Sep 17 00:00:00 2001
From: oliverwen <oliverwen@google.com>
Date: Wed, 13 Jul 2016 16:32:40 -0700
Subject: [PATCH] CHROMIUM: smart antenna algorithm: verification: dbg msg
 priority

 - adjust debug level assigned to certain smart antenna dbg msg

TEST=echo 0x4000 > /sys/module/ath10k_core/parameters/debug_mask,
echo 1 > /sys/kernel/debug/.../ath10k/smart_antenna/smart_ant_debug_level,
debug message "Smart antenna debug level is set to be 1" and those
smart antenna debug messages with smart_antenna_debug_level=1 should
be logged in /var/log/messages

BUG=chrome-os-partner:51964
CQ-DEPEND=CL:341556

Signed-off-by: Oliver Wen <oliverwen@chromium.org>
(cherry picked from wireless-3.18)

Change-Id: I572b1e19aaf77261d2029f5c68168c0982377c3e
Reviewed-on: https://chromium-review.googlesource.com/341557
Commit-Ready: Yixiang Li <yixiang@google.com>
Tested-by: Oliver Wen <oliverwen@chromium.org>
Reviewed-by: Yixiang Li <yixiang@google.com>
---
 .../net/wireless-4.2/ath/ath10k/smart_ant.c   | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/drivers/net/wireless/ath/ath10k/smart_ant.c b/drivers/net/wireless/ath/ath10k/smart_ant.c
index 0d58c3a2d18..773738f56b3 100644
--- a/drivers/net/wireless/ath/ath10k/smart_ant.c
+++ b/drivers/net/wireless/ath/ath10k/smart_ant.c
@@ -585,7 +585,7 @@ static u8 smart_ant_sel_rx_ant(struct ath10k *ar,
 	nsta_sel_ant = sa_info->num_sta_per_ant[tinfo->sel_ant];
 
 	if (ar->smart_ant_info.debug_level >=
-	    ATH10K_SMART_ANT_DBG_LVL_TRAIN_STATES) {
+	    ATH10K_SMART_ANT_DBG_LVL_TRAIN_STAGES) {
 		ath10k_dbg(ar, ATH10K_DBG_SMART_ANT,
 			   "sa_info->default_ant = %d, tinfo->sel_ant= %d\n sa_info->num_sta_conneted = %d, nsta_sel_ant = %d\n",
 			   sa_info->default_ant, tinfo->sel_ant,
@@ -923,8 +923,11 @@ static u8 smart_ant_proc_train_stats(struct ath10k *ar,
 		if (ar->smart_ant_info.debug_level >=
 		    ATH10K_SMART_ANT_DBG_LVL_TOP_DECISION) {
 			ath10k_dbg(ar, ATH10K_DBG_SMART_ANT,
-				   "Training completed:antenna selected %d\n",
-				   tstats->ant_map[0]);
+				   "Training completed:antenna selected %d PER %d RSSI %d %d %d rate %x\n",
+				   tstats->ant_map[0],
+				   tstats->per,
+				   tstats->rssi[0][0], tstats->rssi[1][0],
+				   tstats->rssi[2][0], tdata->rate_code);
 		}
 		/* force config tx/rx antenna after SA decision is made */
 		status |= ATH10K_SMART_ANT_ACT_TX_CFG;
@@ -1039,7 +1042,7 @@ static u8 smart_ant_perf_train_trigger(struct ath10k *ar,
 					ATH10K_SMART_ANT_STATE_PRETRAIN;
 			status = ATH10K_SMART_ANT_ACT_TRAIN;
 			if (ar->smart_ant_info.debug_level >=
-				ATH10K_SMART_ANT_DBG_LVL_TRAIN_STAGES) {
+				ATH10K_SMART_ANT_DBG_LVL_TOP_DECISION) {
 				ath10k_dbg(ar, ATH10K_DBG_SMART_ANT, "start Perf train\n");
 			}
 		}
@@ -1062,7 +1065,7 @@ static u8 smart_ant_retrain_trigger(struct ath10k *ar,
 	if (elapsed_ts > sparams.retrain_interval ||
 	    sa_sta->train_info.train_start) {
 		if (ar->smart_ant_info.debug_level >=
-		    ATH10K_SMART_ANT_DBG_LVL_TRAIN_STAGES) {
+		    ATH10K_SMART_ANT_DBG_LVL_TOP_DECISION) {
 			ath10k_dbg(ar, ATH10K_DBG_SMART_ANT,
 				   "start Periodic train\n");
 		}
@@ -1270,7 +1273,7 @@ static int smart_ant_config_tx(struct ath10k *ar,
 	smart_ant_get_tx_ant(sa_sta, cfg->tx_ant_cfg.tx_ants);
 
 	if (ar->smart_ant_info.debug_level >=
-	    ATH10K_SMART_ANT_DBG_LVL_TRAIN_STAGES) {
+	    ATH10K_SMART_ANT_DBG_LVL_TOP_DECISION) {
 		ath10k_dbg(ar, ATH10K_DBG_SMART_ANT,
 			   "Configure tx antenna for %pM to %d\n",
 			   cfg->tx_ant_cfg.peer_mac,
@@ -1300,7 +1303,7 @@ static int smart_ant_config_rx(struct ath10k *ar,
 	cfg->rx_ant_cfg.rx_ant = info->default_ant;
 
 	if (ar->smart_ant_info.debug_level >=
-	    ATH10K_SMART_ANT_DBG_LVL_TRAIN_STAGES) {
+	    ATH10K_SMART_ANT_DBG_LVL_TOP_DECISION) {
 		ath10k_dbg(ar, ATH10K_DBG_SMART_ANT,
 			   "Configure Rx antenna to %d\n",
 			   cfg->rx_ant_cfg.rx_ant);
@@ -1759,7 +1762,7 @@ int ath10k_smart_ant_sta_connect(struct ath10k *ar,
 	ether_addr_copy(arg.mac_addr.addr, sta->addr);
 
 	if (ar->smart_ant_info.debug_level >=
-	    ATH10K_SMART_ANT_DBG_LVL_TOP_DECISION) {
+	    ATH10K_SMART_ANT_DBG_LVL_TRAIN_STAGES) {
 		ath10k_dbg(ar, ATH10K_DBG_SMART_ANT,
 			   "%s mac %pM vdev_id %d num_cfg %d\n",
 			   __func__,
-- 
2.43.0

